src/util/numeric.h

```diff
@@ -34,6 +33,23 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 	y = temp; \
 } while (0)
 
+// NOTE: decltype forces it to use type of first argument
+template <typename T>
+T rangelim(T value, decltype(value) min, decltype(value) max)
+{
+	if (value < min)
+		return min;
+	if (value > max)
+		return max;
+	return value;
+}
+
+// limits value to range of type T
+template <typename T>
+T rangelim(long value)
```

Comment:
How about floating point? `T value` seems more appropriate, maybe even with const ref, as the value is never modified.
Also, `inline`?

Reply:
Immediate value is faster. `inline` is sensible but entirely optional, compiler will do that anyway.